src/main/java/net/sf/jabref/logic/importer/MrDLibFetcher.java

```diff
@@ -0,0 +1,302 @@
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.ssl.SSLContexts;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.DefaultHandler;
+
+/**
+ *
+ *
+ */
+public class MrDLibFetcher implements SearchBasedFetcher {
+
+    private static final String NAME = "Mr. DLib Fetcher";
+    private List<BibEntry> bibEntryList;
+    private final BibEntry selectedEntry;
+    private List<String> htmlSnippets;
+    private static final Log LOGGER = LogFactory.getLog(MrDLibFetcher.class);
+
+
+    public MrDLibFetcher(BibEntry selectedEntry) throws Exception {
+        this.selectedEntry = selectedEntry;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public List<BibEntry> performSearch(String query) throws FetcherException {
+        String response = makeServerRequest();
+        bibEntryList = convertToBibEntry(response);
+        htmlSnippets = convertToHtml(response);
+        return bibEntryList;
+    }
+
+    /**
+     * Contact the server with the title of the selected item
+     * @param query
+     * @return
+     */
+    private String makeServerRequest() {
+        String response = "";
+
+        //Makes a request to the RESTful MDL-API. Example document.
+        //Servers-side functionality in implementation.
+
+        SSLContext sslcontext = null;
+
+        try {
+            sslcontext = SSLContexts.custom().loadTrustMaterial(null, new TrustSelfSignedStrategy()).build();
+        } catch (KeyManagementException | NoSuchAlgorithmException | KeyStoreException e1) {
+            LOGGER.error(e1.getMessage(), e1);
+        }
+
+
+        SSLConnectionSocketFactory sslsf = new SSLConnectionSocketFactory(sslcontext);
+        try (CloseableHttpClient httpclient = HttpClients.custom().setSSLSocketFactory(sslsf).build()) {
+        Unirest.setHttpClient(httpclient);
+
+            String query = selectedEntry.getLatexFreeField("title").get();
+        try {
+            response = Unirest
+                        .get("https://api-dev.mr-dlib.org/v1/documents/gesis-smarth-0000000300/related_documents/")
+                    .asString().getBody();
+        } catch (UnirestException e) {
+                LOGGER.error(e.getMessage(), e);
+        }
+
+        //Conversion. Server delivers false format, conversion here, TODO to fix
+        response = response.replaceAll("&gt;", ">");
+        response = response.replaceAll("&lt;", "<");
+        } catch (IOException e1) {
+            LOGGER.error(e1.getMessage(), e1);
+        }
+        return response;
+    }
+
+    /**
+     * Converts the xml document responded by the mdl server to an bib entry. Is there a parser somewhere?
+     * @param xmlDocument
+     * @return
+     */
+    private List<BibEntry> convertToBibEntry(String recomendations) {
```

Comment:
I think the `net.sf.jabref.logic.importer.Parser` interface is even more appropriate since it ignores the metadata-part by design. I think the `GvkParser` comes the closest to what you like to do.